unlock-app/src/services/walletService.js

```diff
@@ -26,15 +26,25 @@ export const keyId = (lock, owner) => [lock, owner].join('-')
  * actually retrieving the data from the chain/smart contracts
  */
 export default class WalletService extends EventEmitter {
-  constructor(availableProviders = providers) {
+  constructor(
+    availableProviders = providers,
+    timeout = setTimeout,
```

Comment:
This is very very strange. I am not sure this is the right approach. Are jest time mocks working?

Reply:
no, jest time mocks do not work. at all.